libnmstate/validator.py

```diff
@@ -170,9 +170,18 @@ def validate_dns(state):
     dns_servers = (
         state.get(DNS.KEY, {}).get(DNS.CONFIG, {}).get(DNS.SERVER, [])
     )
-    if len(dns_servers) > 2:
+    if len(dns_servers) > 3:
+        logging.warning(
+            "The libc resolver may not support more than 3 nameservers."
+        )
+    if (
+        len(dns_servers) > 2
+        and any(n for n in dns_servers if is_ipv6_address(n))
+        and any(n for n in dns_servers if not is_ipv6_address(n))
+    ):
         raise NmstateNotImplementedError(
-            "Nmstate only support at most 2 DNS name servers"
+            "Nmstate does support 3 or more mixed "
+            "IPv4 and IPv6 name servers yet."
```

Comment:
The message is not completely correct since the negation is missing "does not". This is another approach to write it without negation:

```suggestion
            "Three or more names servers are only supported when using either IPv4 or IPv6 nameservers."
```

Reply:
@tyll, I think that to your suggestion you need to add at the end something like "but not both".